client/src/PublicRequest.ts

```diff
@@ -178,6 +185,73 @@ export type SignBtcTransactionRequest
     = SignBtcTransactionRequestStandard
     | SignBtcTransactionRequestCheckout;
 
+export type SignSwapRequest = SimpleRequest & {
+    swapId: string,
+    fund:
+        {type: 'NIM'}
+            & Omit<TransactionInfo,
+                'senderType'
+                | 'recipient'
+                | 'recipientType'
+                | 'recipientLabel'
+                | 'validityStartHeight'
+                | 'data'
+                | 'flags'
+            >
```

Comment:
We don't need quite a bit of information for only displaying the swap info visually, that's why it's all omitted here.